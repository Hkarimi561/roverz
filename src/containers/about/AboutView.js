import React from 'react';
import {
  StyleSheet,
  Text,
  Image,
  View,
} from 'react-native';

import { AppStyles, AppColors } from '../../theme';
import Application from '../../constants/config';

const styles = StyleSheet.create({
  paraText: {
    fontSize: 14,
    fontFamily: 'OpenSans-Regular',
    color: 'rgba(255,255,255,0.7)',
  },
  paraTextLight: {
    fontSize: 14,
    fontFamily: 'OpenSans-Regular',
    color: 'rgba(255,255,255,0.5)',
  },
  headText: {
    fontSize: 16,
    fontFamily: 'OpenSans-Regular',
    color: 'rgba(255,255,255,1)',
  },
});

export default class AboutView extends React.Component {
  render() {
    return (
      <View
        style={{
          flex: 1,
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center',
          backgroundColor: AppColors.brand().secondary,
          padding: 15,
        }}
      >
        <View
          style={{
            flex: 1,
            justifyContent: 'flex-end',
          }}
        >
          <Image
            source={Application.logo}
            style={{
              opacity: 1,
              width: 150,
              resizeMode: 'contain',
            }}
          />
        </View>
        <View style={[AppStyles.windowSize, {
          flex: 1,
          flexDirection: 'column',
          justifyContent: 'space-between',
          alignItems: 'center',
        }]}
        >
          <View style={{ alignItems: 'center' }}>
            <Text
              style={[styles.headText]}
            >{`Version: ${Application.aboutDetails.version}`}</Text>
            <Text
              style={[styles.paraText]}
            >{`Build: ${Application.aboutDetails.build}`}</Text>
          </View>
          <View style={{ alignItems: 'center' }}>
            <Text
              style={[styles.paraText]}
            >{Application.aboutDetails.website}</Text>
            <Text
              style={[styles.paraText]}
            >{Application.aboutDetails.email}</Text>
          </View>
          <View>
            <Text
              style={[styles.paraTextLight]}
            >{Application.aboutDetails.company}</Text>
          </View>
        </View>
      </View>
    );
  }
}

AboutView.defaultProps = {
};

AboutView.propTypes = {
};
