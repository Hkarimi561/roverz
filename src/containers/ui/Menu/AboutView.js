import React from 'react';
import {
  StyleSheet,
  Text,
  Image,
  View,
} from 'react-native';

import { AppStyles, AppColors } from '@theme';

const styles = StyleSheet.create({
  paraText: {
    fontSize: 14,
    fontFamily: 'OpenSans-Regular',
    color: 'rgba(255,255,255,0.7)',
  },
  paraTextLight: {
    fontSize: 14,
    fontFamily: 'OpenSans-Regular',
    color: 'rgba(255,255,255,0.5)',
  },
  headText: {
    fontSize: 16,
    fontFamily: 'OpenSans-Regular',
    color: 'rgba(255,255,255,1)',
  },
});

export default class AboutView extends React.Component {
  render() {
    return (
      <View
        style={{
          flex: 1,
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center',
          backgroundColor: AppColors.brand.secondary,
          padding: 15,
        }}
      >
        <View
          style={{
            flex: 1,
            justifyContent: 'flex-end',
          }}
        >
          <Image
            source={require('../../../images/logo.png')} // eslint-disable-line global-require
            style={{
              opacity: 1,
              width: 150,
              resizeMode: 'contain',
            }}
          />
        </View>
        <View style={[AppStyles.windowSize, {
          flex: 1,
          flexDirection: 'column',
          justifyContent: 'space-between',
          alignItems: 'center',
        }]}
        >
          <View style={{ alignItems: 'center' }}>
            <Text
              style={[styles.headText]}
            >Version: 2.0.1</Text>
            <Text
              style={[styles.paraText]}
            >Build: 1020</Text>
          </View>
          <View style={{ alignItems: 'center' }}>
            <Text
              style={[styles.paraText]}
            >www.mongrov.com</Text>
            <Text
              style={[styles.paraText]}
            >support@mongrov.com</Text>
          </View>
          <View>
            <Text
              style={[styles.paraTextLight]}
            >Powered by Mongrov, Inc.</Text>
          </View>
        </View>
      </View>
    );
  }
}

AboutView.defaultProps = {
};

AboutView.propTypes = {
};
