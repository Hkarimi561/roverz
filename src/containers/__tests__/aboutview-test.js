/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import AboutView from '@containers/about/AboutView';

it('AboutView renders correctly', () => {
  const tree = renderer.create(
    <AboutView />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
