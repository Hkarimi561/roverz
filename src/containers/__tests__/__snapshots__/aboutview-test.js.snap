// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#50527F",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <Image
      source={null}
      style={
        Object {
          "opacity": 1,
          "resizeMode": "contain",
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "height": 1334,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,1)",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,1)",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
            },
          ]
        }
      >
        Version: undefined
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.7)",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 14,
            },
          ]
        }
      >
        Build: undefined
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "borderRadius": 2,
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#373856",
                },
                Object {
                  "borderRadius": 2,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 8,
                  "paddingHorizontal": 20,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#fff",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    Object {
                      "fontSize": 12,
                    },
                    undefined,
                    Object {
                      "fontWeight": "600",
                    },
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                  ],
                ]
              }
            >
              Help
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.7)",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 14,
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.7)",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 14,
            },
          ]
        }
      />
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.5)",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 14,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
